sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1240,7 +1286,8 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
             or shape (n_categorical_features,), default=None
         Indicates the categorical features.
 
-        - None : no feature will be considered categorical.
+        - None : no feature will be considered categorical unless explicitly
```

Comment:
I am wondering if we should keep `None` as is and have an option `"auto"` that could become the default in 2 versions. `None` could be a bit less explicit than a `str` option (I proposed "auto" but it could be something more linked to the "categorical" dtype from pandas).

Just a thought.

Reply:
We could as well avoid the delay of 2 versions since I don't think that we have a corner case where our user can be surprised.

Reply:
Has I said here (https://github.com/scikit-learn/scikit-learn/pull/24907#issuecomment-1312800430), since this case would previously raise an exception, I don't really see the point of trying to enforce backward compat.